modules/compute/gke-node-pool/gpu_direct.tf

```diff
@@ -0,0 +1,46 @@
+/**
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+# Enable GPUDirect for A3 and A3Mega VMs, this involve multiple kubectl steps to integrate with the created cluster
+# 1. Install NCCL plugin daemonset
+# 2. Install NRI plugin daemonset
+# 3. Update user workload to inject rxdm sidecar and other required annotation, volume etc.
+locals {
+  user_workload_path_tcpx  = var.user_workload_path == null ? "${path.module}/gpu-direct-workload/sample-tcpx-workload-job.yaml" : "${path.module}/${var.user_workload_path}"
+  user_workload_path_tcpxo = var.user_workload_path == null ? "${path.module}/gpu-direct-workload/sample-tcpxo-workload-job.yaml" : "${path.module}/${var.user_workload_path}"
+
+  gpu_direct_settings = {
+    "a3-highgpu-8g" = {
+      # Manifest to be installed for enabling TCPX on a3-highgpu-8g machines
+      gpu_direct_manifests = [
+        "https://raw.githubusercontent.com/GoogleCloudPlatform/container-engine-accelerators/fee883360a660f71ba07478db95d5c1325322f77/gpudirect-tcpx/nccl-tcpx-installer.yaml",      # nccl_plugin v3.1.9 for tcpx
+        "https://raw.githubusercontent.com/GoogleCloudPlatform/container-engine-accelerators/fee883360a660f71ba07478db95d5c1325322f77/gpudirect-tcpx/nccl-config.yaml",              # nccl_configmap
+        "https://raw.githubusercontent.com/GoogleCloudPlatform/container-engine-accelerators/fee883360a660f71ba07478db95d5c1325322f77/nri_device_injector/nri-device-injector.yaml", # nri_plugin
+      ]
+      updated_user_workload_path = replace(local.user_workload_path_tcpx, ".yaml", "-tcpx.yaml")
+      rxdm_version               = "v2.0.12" # matching nccl-tcpx-installer version v3.1.9
```

Comment:
nccl-tcpx-installer currently is in the sample job, so a user job might require a different rxdm version. Can we take this as an input (with default to 2.0.12) ?

Reply:
There are 2 options here:
1. always get the latest supported nccl-installer version from [container-engine-accelerators](https://github.com/GoogleCloudPlatform/container-engine-accelerators/tree/master), and whenever nccl-installer version updated there, the rxdm version will need to be updated in the cluster toolkit repo to match that.
2. get a fixed version of nccl-installer, and we can decide when to get the newer version and also update the rxdm version to match that.

I prefer option 2 which give us better flexibility, this PR is also configured this way. 
If user have a specific requirement of nccl version, then they would need to update the gpu_direct manifests and rxdm version. I would prefer to hide the rxdm version inside the module because updating rxdm version itself wouldn't solve the problem.

Reply:
I believe that being able to set this in the blueprint will be required so that we can up-level approved version combinations into the blueprint.

How is `fee883360a660f71ba07478db95d5c1325322f77` mapped to `nccl_plugin v3.1.9 for tcpx`?

My ideal case would be as follows:
- Take an input var object named `gpu_direct_versions` example
```
settings:
  gpu_direct_versions:
    rxdm_version: v2.0.12
    container-engine-accelerators_commit: fee883360a660f71ba07478db95d5c1325322f77
```

Ideally instead of `container-engine-accelerators_commit` you could take `nccl_plugin`, hence why I ask how these are mapped.

- Use your proposed combination as default.
- Validate that the two input versions match a compatible list, even if that list is incomplete. For example you could start by having the validation just validate against this single know good version which you have proposed in your pr. Use a [check](https://developer.hashicorp.com/terraform/language/v1.5.x/checks) to perform the validation so that it does not block terraform execution. It will only generate a warning.

In this way we can propose the known good default, warn ppl from leaving the known good versions, but still allow value to be overwritten. This will be important for our own testing and also for supplying multiple known good versions as inputs in reference designs.

Reply:
added warning in the output instruction about the version mapping